### ISSUE TITLE
Shut down on Window close not working

### ISSUE THREAD

**Issue by Cpt-Dingus** [wwmm/easyeffects#1423]
### EasyEffects Version

6.2.3

### What package are you using?

Flatpak (Flathub)

### Distribution

Kubuntu 20.04

### Describe the bug

Clicking options, trying to toggle a "Service" option (That being "Shut down on window close" and "Start service on login") will make the slider jump, but not actually toggle - it remains in it's original state. Ie trying to disable "Shut down on window close" will just make the slider jump, but it ultimately returns back to it's original position by itself - on.

### Expected Behavior

The "Service" options should be toggle-able. 
![image](https://user-images.githubusercontent.com/100243410/157182680-2746e834-146a-40dd-8509-dea859241f06.png)

### Debug Log

<details><summary>Debug Log</summary><br/>

```
Paste your log here
```
</details>

**Comment by wwmm** [comment]
We have always had some kind of difficulty with this in the Flatpak packages. @vchernin how are these buttons reacting in your computer?

**Comment by vchernin** [comment]
Oh right, I forgot to reply to this.

This is just missing a user friendly warning (that I will add eventually...), @Cpt-Dingus do you recall being prompted to give EasyEffects background access but then denying it?

If that was the case `flatpak permission-reset com.github.wwmm.easyeffects` will let EasyEffects ask again. Otherwise you're probably missing a dependency.

**Comment by Cpt-Dingus** [comment]
Hello, there was no warning before or after running the `flatpak permission-reset com.github.wwmm.easyeffects` command. How can I check for missing dependencies?

**Comment by vchernin** [comment]
What is outputted when running `flatpak run com.github.wwmm.easyeffects` and trying the switches?

**Comment by vchernin** [comment]
do you have xdg-desktop-portal-kde and xdg-desktop-portal installed?

**Comment by Cpt-Dingus** [comment]
The output of the flatpak command is as following
`You have PipeWire 0.3.48 installed
This is newer or the same as PipeWire 0.3.31 required to run EasyEffects`
And yes, I do have both of these packages installed.

**Comment by Cpt-Dingus** [comment]
I have ran easyeffects via cli and have found this out:
`(easyeffects:2): easyeffects-WARNING **: 09:03:48.807: lv2_wrapper: Could not find the plugin: http://drobilla.net/plugins/mda/Loudness

(easyeffects:2): easyeffects-WARNING **: 09:03:48.808: lv2_wrapper: http://drobilla.net/plugins/mda/Loudness port symbol not found: loudness

(easyeffects:2): easyeffects-WARNING **: 09:03:48.808: lv2_wrapper: http://drobilla.net/plugins/mda/Loudness port symbol not found: output

(easyeffects:2): easyeffects-WARNING **: 09:03:48.808: lv2_wrapper: http://drobilla.net/plugins/mda/Loudness port symbol not found: link

(easyeffects:2): easyeffects-WARNING **: 09:03:49.566: lv2_wrapper: Could not find the plugin: http://drobilla.net/plugins/mda/Loudness

(easyeffects:2): easyeffects-WARNING **: 09:03:49.566: lv2_wrapper: http://drobilla.net/plugins/mda/Loudness port symbol not found: loudness

(easyeffects:2): easyeffects-WARNING **: 09:03:49.566: lv2_wrapper: http://drobilla.net/plugins/mda/Loudness port symbol not found: output

(easyeffects:2): easyeffects-WARNING **: 09:03:49.566: lv2_wrapper: http://drobilla.net/plugins/mda/Loudness port symbol not found: link

(easyeffects:2): easyeffects-WARNING **: 09:03:59.864: portal: a background request failed: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.Background” on object at path /org/freedesktop/portal/desktop

(easyeffects:2): easyeffects-WARNING **: 09:03:59.864: portal: background portal access has likely been denied

(easyeffects:2): easyeffects-WARNING **: 09:03:59.864: portal: to let EasyEffects ask for the portal again, run flatpak permission-reset com.github.wwmm.easyeffects

(easyeffects:2): easyeffects-WARNING **: 09:03:59.864: portal: setting shutdown on window close state and switch to true`

**Comment by vchernin** [comment]
> GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.Background” on object at path /org/freedesktop/portal/desktop

Well, the portal interface is supposed to exist, but I have no idea what could be wrong. This is completly outside control of the applciatioj.

**Comment by Cpt-Dingus** [comment]
Trying ashpd now. The weird thing is that my root partition doesn't even have /org, although I am not sure whether it is supposed to be present or not.
Ashpd didn't change anything, although should I remove xdg-desktop-portal first before trying to re-run easyeffects?

**Comment by vchernin** [comment]
> The weird thing is that my root partition doesn't even have /org, although I am not sure whether it is supposed to be present or not.

My understand is `/org` is not a filesystem directory, rather the way D-bus is namespaced.

> Ashpd didn't change anything, 

So same "No such interface" error?

> although should I remove xdg-desktop-portal first before trying to re-run easyeffects?

I don't see how that would help, it's a required Flatpak dependency, after all.

**Comment by Cpt-Dingus** [comment]
Sorry, I didn't run actually Ashpd lol - pebkac. What am I supposed to do inside of it?

**Comment by vchernin** [comment]
Oh, nvm. It appears xdg-desktop-portal-kde only implemented support for the background portal around the time kubuntu 20.04 was released, so it was never going to work with just xdg-desktop-portal-kde.

https://github.com/KDE/xdg-desktop-portal-kde/commit/0cefb5e18db477be3d2a0d3d05f26ccf186f8d2e

**Comment by vchernin** [comment]
> Sorry, I didn't actually run Ashpd lol - pebkac. What am I supposed to do inside of it?

Shouldn't matter now.

As a workaround could you try installing `xdg-desktop-portal-gtk`?

**Comment by Cpt-Dingus** [comment]
xdg-desktop-portal-gtk sadly didn't change anything. 
How can I do the aforementioned modification to allow support? Not exactly sure what path the files are in.

**Comment by vchernin** [comment]
Did you restart after installing it? It may be required in this case.

> How can I do the aforementioned modification to allow support? Not exactly sure what path the files are in.

It is not just a matter of adding files, for this to work the actual code needs to exist in the right place. The easiest solution will be to wait for kubuntu 22.04 which should definitely work properly.

**Comment by Cpt-Dingus** [comment]
I did restart, still throws the same error even in cli. I'll just wait for 22.04 though, thanks for your help! I'd be stuck in this rabbit hole for ages lol.

**Comment by vchernin** [comment]
> I'd be stuck in this rabbit hole for ages lol.

No worries lol, happy to help.

And for a temporary workaround, [this should be a valid way to manually add the autostart file](https://github.com/flathub/com.github.wwmm.easyeffects/issues/3#issuecomment-908632842). (The switches won't be affected).

**Comment by Cpt-Dingus** [comment]
Yeah, I just added easyeffects to autostart via the native app - all good there.

**Comment by huandney** [comment]
I was having two related problems, I couldn't start it with the system and I couldn't make it run in the background. The problem was that I had installed Easyeffects from my system's app store (Pop!_shop). Installing this way I couldn't toggle the "Service" options should be toggle-able.

After installing `xdg-desktop-portal-gnome` and restarting, I was able to switch, but after closing the window it didn't run in the background and when the system started easyeffect it didn't start together.

The problem of not being able to keep it after closing the window was only solved after a manual installation:
```
flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
flatpak install flathub com.github.wwmm.easyeffects
```
Already the problem of booting with [this](https://github.com/flathub/com.github.wwmm.easyeffects/issues/3#issuecomment-908632842).
